### ISSUE TITLE
Xpress.jl 0.12.0 produces no output

### ISSUE THREAD

**Issue by jjstickel** [jump-dev/Xpress.jl#106]
As the subject says, no output is produced with Xpress.jl 0.12.0. Setting the model as
 `model = Model(() -> Xpress.Optimizer(OUTPUTLOG=1))`
provides verbose output in 0.11.1 (including the annoying "Reading Problem" block), but the same code produces no output for 0.12.0.

**Comment by kdheepak** [comment]
Just for sanity checking, @jjstickel, can you find this file on your machine `/path/to/xpress-install/xpressmp/include/xprs.h`, and search for `OUTPUTLOG` in that file and paste the corresponding line here? 

For me that file is located over here: `~/Applications/fico-xpress/Xpress Workbench.app/Contents/Resources/xpressmp/include/xprs.h`.

**Comment by jjstickel** [comment]
There are a few `#define` lines for variables with OUTPUTLOG in the name. Then there is this block:
```
/***************************************************************************\
 * values related to OUTPUTLOG                                             *
\***************************************************************************/
#define XPRS_OUTPUTLOG_NO_OUTPUT                0
#define XPRS_OUTPUTLOG_FULL_OUTPUT              1
#define XPRS_OUTPUTLOG_ERRORS_AND_WARNINGS      2
#define XPRS_OUTPUTLOG_ERRORS                   3
```
Is that what you are looking for?

**Comment by joaquimg** [comment]
Nope, is there a define
`#define XPRS_OUTPUTLOG`
with no suffix?

**Comment by jjstickel** [comment]
Sounds like you want this:
`#define XPRS_OUTPUTLOG                                               8035`
For completeness, these are the other lines:
`#define XPRS_MSP_OUTPUTLOG                                           6214`
`#define XPRS_MSE_OUTPUTLOG                                           6610`

**Comment by joaquimg** [comment]
Thanks! I found the error, just taking a little longer because there is an intermittent related error, PR coming

**Comment by jjstickel** [comment]
Thanks for fixing the output log. However, the initial "Reading Problem" block of output is back again, even with:
`model = Model(()->Xpress.Optimizer(OUTPUTLOG=0))`
I expect to send output to files now, and so I can probably ignore this behavior.  Just thought I would mention it.

**Comment by joaquimg** [comment]
Are you in 0.12.1?

I just tested on my side and there is no message from Xpress after:
`model = Model(()->Xpress.Optimizer(OUTPUTLOG=0));`

**Comment by jjstickel** [comment]
@joaquimg,
```
$ julia --project

<snip>   
  | | |_| | | | (_| |  |  Version 1.5.2 (2020-09-23)
 
julia> using JuMP;

julia> import Xpress;
[ Info: Xpress: Found license file /Applications/FICO Xpress/xpressmp/bin/xpauth.xpr
[ Info: Xpress: Development license detected.

julia> model = Model(()->Xpress.Optimizer(OUTPUTLOG=0));

Reading Problem
Problem Statistics
           0 (      0 spare) rows
           0 (      0 spare) structural columns
           0 (      0 spare) non-zero elements
Global Statistics
           0 entities        0 sets        0 set members

(testing) pkg> st
Status `~/programming/julia/testing/Project.toml`

<snip>
  [4076af6c] JuMP v0.21.4
  [9e70acf3] Xpress v0.12.1

(testing) pkg>
```

**Comment by joaquimg** [comment]
Linux or mac?

**Comment by jjstickel** [comment]
Mac. Julia installed via Macports, although Julia packages are fetched from git just like other OS's. Xpress is installed via .dmg and moved to /Applications.

**Comment by joaquimg** [comment]
can you test: https://github.com/jump-dev/Xpress.jl/pull/108 ?

**Comment by joaquimg** [comment]
@kdheepak are you on a mac also? Linux should be equally good for testing...
Windows is a bit different.

**Comment by jjstickel** [comment]
Now I get even more output:
```
  [9e70acf3] Xpress v0.12.1 `https://github.com/jump-dev/Xpress.jl.git#jg/log`

julia> model = Model(()->Xpress.Optimizer(OUTPUTLOG=0))

Reading Problem
Problem Statistics
           0 (      0 spare) rows
           0 (      0 spare) structural columns
           0 (      0 spare) non-zero elements
Global Statistics
           0 entities        0 sets        0 set members
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: Xpress
```

**Comment by joaquimg** [comment]
For the additional output it is just a matter of adding ; in the of the line. If that was in the middle of a scrip the addtional stuff would not show

**Comment by jjstickel** [comment]
> For the additional output it is just a matter of adding ; in the of the line. If that was in the middle of a scrip the addtional stuff would not show

Of course, you are right. I edited my comment to show the relevant part.

**Comment by kdheepak** [comment]
Why do we have `log_level != 0 && ` in these lines?

https://github.com/jump-dev/Xpress.jl/blob/1e563eb81895817a449a866d5622e10fa70ab0fc/src/MOI/MOI_wrapper.jl#L300-L304

**Comment by kdheepak** [comment]
Okay, I figured out what was going on. I think we should apply this patch.

```
diff --git a/src/MOI/MOI_wrapper.jl b/src/MOI/MOI_wrapper.jl
index ac3c9ee..90cd423 100644
--- a/src/MOI/MOI_wrapper.jl
+++ b/src/MOI/MOI_wrapper.jl
@@ -266,13 +266,13 @@ mutable struct Optimizer <: MOI.AbstractOptimizer
 
         model.inner = XpressProblem()
 
-        # get default log level
-        model.log_level = MOI.get(model, MOI.RawParameter("OUTPUTLOG"))
-
         for (name, value) in model.params
             MOI.set(model, name, value)
         end
 
+        # get default log level
+        model.log_level = MOI.get(model, MOI.RawParameter("OUTPUTLOG"))
+
         model.variable_info = CleverDicts.CleverDict{MOI.VariableIndex, VariableInfo}()
         model.affine_constraint_info = Dict{Int, ConstraintInfo}()
         model.sos_constraint_info = Dict{Int, ConstraintInfo}()
```

This is the behavior we would get in that case

![Screen Shot 2020-11-19 at 6 30 26 PM](https://user-images.githubusercontent.com/1813121/99744621-4caf9080-2a95-11eb-98f2-4ebc9f65155b.png)

Thoughts?

**Comment by joaquimg** [comment]
I fixed #108 and tested it on Linux, seems to be ok on CI also. Let me know if you find issues. You can test on master
